connect/rest-extension/src/main/java/org/apache/kafka/connect/rest/extension/JaasBasicAuthFilter.java

```
@@ -0,0 +1,93 @@
+                new LoginContext(CONNECT_LOGIN_MODULE, new BasicAuthCallBackHandler(
+                    requestContext.getHeaderString(AUTHORIZATION)));
+            loginContext.login();
+        } catch (LoginException e) {
+            requestContext.abortWith(
+                Response.status(Response.Status.UNAUTHORIZED)
+                    .entity("User cannot access the resource.")
+                    .build());
+        }
+    }
+
+
+    public static class BasicAuthCallBackHandler implements CallbackHandler {
+
+        private String userName;
+        private String password;
+
+        public BasicAuthCallBackHandler(String credentials) {
+            if (credentials != null) {
+                int space = credentials.indexOf(32);
+                if (space > 0) {
+                    String method = credentials.substring(0, space);
+                    if ("basic".equalsIgnoreCase(method)) {
+                        credentials = credentials.substring(space + 1);
+                        credentials = new String(Base64.decoder().decode(credentials),
+                                                 StandardCharsets.UTF_8);
+                        int i = credentials.indexOf(":");
+                        if (i > 0) {
+                            userName = credentials.substring(0, i);
+                            password = credentials.substring(i + 1);
+                        }
+                    }
+                }
+            }
+        }
+
+        @Override
+        public void handle(Callback[] callbacks) throws IOException, UnsupportedCallbackException {
+            for (Callback callback : callbacks) {
+                if (callback instanceof NameCallback) {
+                    ((NameCallback) callback).setName(userName);
+                } else if (callback instanceof PasswordCallback) {
+                    ((PasswordCallback) callback).setPassword(password.toCharArray());
+                }
```

Should we throw `UnsupportedCallbackException` if it doesn't match? Especially since we declare it and that's the intended use of this exception according to the interface (actually we're violating the interfaces contract if we don't).